src/__tests__/getStepStartStates.test.ts

```diff
@@ -48,7 +48,8 @@ test('should throw if "scopes" not provided in execution config', () => {
     partialInstanceConfig: { scopes: undefined },
   });
 
-  delete context.instance.config.scopes;
+  // Latest version of TS do not allow deleting properties that are not optional
+  delete (context as any).instance.config.scopes;
```

Comment:
```suggestion
  if (context?.instance?.config?.scopes) context.instance.config.scopes = undefined;
```
delete isn't implemented that well in JS and should be avoided if possible

Reply:
Ya I avoid `delete` at all costs. The purpose of this test is to to test cases where the property is not present at all. Not even assigned to `undefined`.